File: 1770-minimum-deletions-to-make-character-frequencies-unique/minimum-deletions-to-make-character-frequencies-unique.py
class Solution:
    def minDeletions(self, s: str) -> int:
        #count the letter and and check freq is differnt
        count=Counter(s) # a-->3 ..
        res=0
        used_freq=set()
        for letter,freq in count.items():
            while freq>0 and freq in used_freq:
                freq-=1#delete it
                res+=1 
            used_freq.add(freq) #exclude 0 
        
        return res

        

File: 1770-minimum-deletions-to-make-character-frequencies-unique/minimum-deletions-to-make-character-frequencies-unique.js
/**
 * @param {string} s
 * @return {number}
 */
var minDeletions = function(s) {
     const chars = new Map();
    for (const char of s) {
        chars.set(char, (chars.get(char) || 0) + 1);
    }

    const freqSet = new Set();
    let count = 0;

    for (let freq of chars.values()) {
        while (freq > 0 && freqSet.has(freq)) {
            freq--;
            count++;
        }

      freqSet.add(freq);
    }

    return count;    
};
